[SPARK-28473][DOC] Stylistic consistency of build command in README

Mister-Meeseeks opened the pull request:
## What changes were proposed in this pull request?

Change the format of the build command in the README to start with a `./` prefix

    ./build/mvn -DskipTests clean package

This increases stylistic consistency across the README- all the other commands have a `./` prefix. Having a visible `./` prefix also makes it clear to the user that the shell command requires the current working directory to be at the repository root. 

## How was this patch tested?

README.md was reviewed both in raw markdown and in the Github rendered landing page for stylistic consistency.

Mister-Meeseeks committed:
Update README.md

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

dongjoon-hyun commented on README.md:
```diff
@@ -25,7 +25,7 @@ This README file only contains basic setup instructions.
 To build Spark and its example programs, run:
 
-    build/mvn -DskipTests clean package
+    ./build/mvn -DskipTests clean package
```
Could you find more instances, @Mister-Meeseeks ? For example, `R/README.md` also has this.

Mister-Meeseeks committed:
Update docs so that all shell command examples are prepended with ./

To enforce stylistic consistency and increase clarity for the user, shell
command examples in the docs were prepended with `./`. For example

    bin/spark-submit --jars ...

becomes

    ./bin/spark-submit --jars ...

This is the convention adhered to for most of the documentation already. This
commit converts the minority of examples that don't.

The `./` prefix is only added in the context of a specific instructional
command. When the docs are discussing in the context of a script, the prefix
is not included. For example "the `bin/spark-submit` script is used to launch
jobs" would *not* require the `./` prefix.

Mister-Meeseeks committed:
Enforce ./-prepended command consistencency in documentation

Mister-Meeseeks commented:
I scoured the documentation for offending examples. I believe I found and corrected every command example not conforming to the `./` prepended convention.

srowen reviewed (commented):
Looks good if we can fix all of them consistently

dongjoon-hyun commented:
ok to test.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13168/
Test PASSed.

SparkQA commented:
**[Test build #108061 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108061/testReport)** for PR 25231 at commit [`53d59fc`](https://github.com/apache/spark/commit/53d59fcd82c77775390bb8142342c7f048893d82).

SparkQA commented:
**[Test build #108061 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108061/testReport)** for PR 25231 at commit [`53d59fc`](https://github.com/apache/spark/commit/53d59fcd82c77775390bb8142342c7f048893d82).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/108061/
Test PASSed.

dongjoon-hyun reviewed (approved):
+1, LGTM.
Thank you, @Mister-Meeseeks and @srowen .
Merged to master.
